src/pool/srv_target.c

```diff
@@ -614,27 +614,29 @@ ds_pool_cache_fini(void)
 	daos_lru_cache_destroy(pool_cache);
 }
 
-struct ds_pool *
-ds_pool_lookup(const uuid_t uuid)
+int
+ds_pool_lookup(const uuid_t uuid, struct ds_pool **pool)
 {
 	struct daos_llink	*llink;
-	struct ds_pool		*pool;
```

Comment:
yes, this is the assumption for this function.  You mean add local variable? I would rather not do that, in case it is really necessary to simplify the logic.  Though I may miss your question? thanks.

Reply:
I was referring to the convention to only set the output argument upon success, so yes I guess that would involve adding a local variable. But, commenting the function  makes the behavior known to the code reader.